usermin/edit_lang.cgi

```diff
@@ -19,6 +19,28 @@ print &ui_table_row($text{'lang_lang'},
            [ map { [ $_->{'lang'}, "$_->{'desc'}" ] }
                  &list_languages() ]));
 
+# Old datetime format or a new locale
+my $locale;
+eval "use DateTime; use DateTime::Locale; use DateTime::TimeZone;";
+if (!$@) {
+        $locale++;
+        my $locales = &list_locales();
+        my %localesrev = reverse %{$locales};
+        my $locale_auto = &parse_accepted_language(\%uconfig);
+        print &ui_table_row($text{'lang_locale'},
+                &ui_select("locale", $locale_auto || $uconfig{'locale'} || &get_default_system_locale(),
+                           [ map { [ $localesrev{$_}, $_ ] } sort values %{$locales} ]).
+                           &ui_hidden("dateformat", $uconfig{'dateformat'}));
+        }
+
+if (!$locale) {
```

Comment:
Instead of setting $locale, why not just make this an `else` block?

Reply:
That's right! Fixed! It was introduced because of nested `eval` tests that I have also fixed.